### PR TITLE
Update Android SDK to Android 11

### PR DESCRIPTION
**Pull request by SanoopC**
* Target SDK version upgraded to 30
* Migrated to AndroidX
* Deprecated GoogleApiClient replaced with FusedLocationProviderClient

### CODE REVIEW

File: sift/src/main/java/siftscience/android/AppStateCollector.java [hunk 1]
```diff
@@ -60,15 +71,11 @@ public AppStateCollector(SiftImpl sift, Context context) {
...
-                        .addApi(LocationServices.API)
-                        .addConnectionCallbacks(this)
-                        .addOnConnectionFailedListener(this)
-                        .build();
-            } catch (Exception e) {
-                Log.e(TAG, e.toString());
-            }
+            mFusedLocationClient = LocationServices.getFusedLocationProviderClient(this.context);
+            mSettingsClient = LocationServices.getSettingsClient(this.context);
+            createLocationCallback();
```
Comment (garylee1) [inline sift/src/main/java/siftscience/android/AppStateCollector.java:76]: is there any reason all of these helpers directly modify instance state versus returning built objects?

**Review by garylee1 — COMMENTED** [review]
LGTM, with one comment

File: sift/src/main/java/siftscience/android/AppStateCollector.java (hunk 1 above)
Comment (SanoopC) [reply sift/src/main/java/siftscience/android/AppStateCollector.java:76]: @garylee1 Those are the initialization of helpers and there is no builder for FusedLocationProviderClient as in the GoogleApiClient. 
We followed as per the guidelines mentioned in [android/location-samples](https://github.com/android/location-samples/blob/02e5fb05c80bf2ffac430bb8a57158356de84d86/LocationUpdates/app/src/main/java/com/google/android/gms/location/sample/locationupdates/MainActivity.java#L185)

**Review by garylee1 — APPROVED** [review]

**Comment by sanogueralorenzo** [comment]
Hello, when is this expected to be merged and released?

Would this close the AndroidX issue described here? https://github.com/SiftScience/sift-android/issues/74 siftscience is currently blocking my current company from dropping jetifier.

**Comment by phanther** [comment]
@sanogueralorenzo We have a future ticket for that fix which is going to be picked up soon, will update on the timeline shortly

**Comment by phanther** [comment]
@sanogueralorenzo just a quick update, I was mistaken earlier and this ticket has the merged code to address AndroidX issue

**Pull request merged**